### ISSUE TITLE
iOS 14+ only: Modal on top of a Dialog is not accessible with voice over

### ISSUE THREAD

**Issue by cvabanuk3** [microsoft/fluentui#17088]
2. This issue is caused by a Fluent UI React control. -  **can be an iOS regression too** as it works fine on iOS <14. 
     Still can be an issue with Fluent too as none of the controls from Modal (the top most one) was not accessible.

### Environment Information

- **Package version(s)**: 
    "@uifabric/icons": "7.5.21",
    "office-ui-fabric-core": "^11.0.0",
    "office-ui-fabric-react": "7.123.1",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-scripts-ts": "2.13.0"
- **Browser and OS versions**: 
- iOS 14+, Mobile or Tablet browsers

### Describe the issue:

Open Dialog from a landing page and then Open Modal from Dialog, and use Voice over on iOS 14+ devices to access Modal.
Voice over can't recognize any of the elements from Modal.

### Please provide a reproduction of the issue in a codebox:
code: https://codesandbox.io/s/fluent-modal-on-dialog-8zkh4?file=/src/index.tsx

You can try accessing this link https://8zkh4.csb.app/ from any iOS 14+ device with voice over and can experience the issue.
the same is working as expected on iOS 13.7 (iPhone 7)

**REPRO**: 
1. Goto https://8zkh4.csb.app/ on a iOS 14+ device
2. start voice over
3. tap to "Open dialog" button
4. double tap to select|click --> shows up a Dialog with close button in right top corner.
5. click(double tap) on "Close" Button
6. shows up Modal with a close button.
7. voice over can't reach out to any of the elements on the Modal.

**Comment by joschect** [comment]
Hey @cvabanuk3 Thanks for the submission! I'm not sure but I believe that this is happening because the first modal is blocking and is likely trying to keep focus. As a work around, you can try something like [this modified version of your codesandbox](https://codesandbox.io/s/fluent-modal-on-dialog-forked-0tg1h?file=/src/index.tsx) which might allow voice over to interact with the second modal.

**Comment by cvabanuk3** [comment]
Thanks @joschect, unfortunately it doesn't work.

**Comment by joschect** [comment]
Hey @cvabanuk3 after consulting with @smhigley , this is an issue with Voice Over. Voice over expects there to be only 1 modal dialog at any time. Moving forward there are a couple of options that you have. You can either, dismiss the initial dialog and then show it again if the users cancels. Or you can change the content of the dialog along with the title. Swapping it out will ensure the correct behavior as well.

**Comment by joschect** [comment]
@smhigley can correct me if I'm wrong but I believe that best practices are to have only one dialog open at any given time.

**Comment by smhigley** [comment]
Yup! I'd definitely suggest only having one modal dialog open at a time. You could switch out content within the same dialog (and remember to handle focus and update the dialog name as @joschect mentioned), or you could open a new dialog and dismiss the old one.

**Comment by cvabanuk3** [comment]
Thanks for the suggestion @joschect  & @smhigley, got a design change approval. Thanks for your time.
